src/ErrorHandler.php

```diff
@@ -111,52 +100,72 @@ private function __construct(callable $callback, int $reservedMemorySize = self:
             // first call to the set_error_handler method would cause the PHP
             // bug https://bugs.php.net/63206 if the handler is not the first
             // one in the chain of handlers
-            set_error_handler([$this, 'handleError'], $this->capturedErrors);
-
-            $this->isRoot = true;
+            set_error_handler([$this, 'handleError'], E_ALL);
         }
 
         $this->previousExceptionHandler = set_exception_handler([$this, 'handleException']);
     }
 
     /**
-     * Registers this error handler and associates the given callback to the
-     * instance.
-     *
-     * @param callable $callback           callback that will be called when exception is caught
-     * @param int      $reservedMemorySize The amount of memory to reserve for the fatal error handler
+     * Gets the current registered error handler; if none is present, it will register it.
      *
-     * @return self
+     * @param int|null $reservedMemorySize The requested amount of memory to reserve
      *
-     * @throws \ReflectionException
+     * @return ErrorHandler The ErrorHandler singleton
      */
-    public static function register(callable $callback, int $reservedMemorySize = self::DEFAULT_RESERVED_MEMORY_SIZE): self
+    public static function getRegisteredHandler(int $reservedMemorySize = null): self
     {
-        return new self($callback, $reservedMemorySize);
+        if (null === self::$registeredHandler) {
+            self::$registeredHandler = new self($reservedMemorySize ?? self::DEFAULT_RESERVED_MEMORY_SIZE);
+        } elseif (null !== $reservedMemorySize) {
+            self::setReservedMemory($reservedMemorySize);
+        }
+
+        return self::$registeredHandler;
     }
 
     /**
-     * Sets the PHP error levels that will be captured by the Raven client when
-     * a PHP error occurs.
+     * Adds a listener to the current error handler to be called upon each invoked captured error;
+     * if no handler is registered, this method will instantiate and register it.
      *
-     * @param int  $levels  A bit field of E_* constants for captured errors
-     * @param bool $replace Whether to replace or amend the previous value
+     * If you want to use a naked callable and not an object as a listener,
```

Comment:
I've removed that line because it was no true. I would prefer the interface to avoid issues that would arise only while handling errors, which would complicate the situation a lot.

I would move to `callable` only if we could validate the function signature, but that would be costly and error prone. @HazAT WDYT?

Reply:
I don't think we should validate the signature since it would slow down the performance for nothing (it's up to the user to provide a valid callable). If in the future something like [HHVM fun](http://docs.hhvm.com/hack/reference/function/HH.fun/) will be available in PHP we will be able to check it. I like the interface concept but I fear that it could be limiting for simple use cases which doesn't require a whole class to be implemented. Also as said in another comment if we go that way then we cannot use `__invoke` or it will be impossible for a class to implement both the `ErrorHandlerInterface` and the `ExceptionHandlerInterface` or the methods will clash

Reply:
I agree on changing `__invoke`; I will change it if we decide to maintain the interfaces, otherwise we're dropping them in favor of callables.

Reply:
@HazAT said that `callable` it's better. I've dropped the interfaces and switched to `callable`s.